### PR TITLE
modify OSS controller that allow users to search download location regardless of https://, http://, www

### PR DESCRIPTION
**Pull request by doggai10**
Signed-off-by: doggai10 <doggai77@gmail.com>

## Description
modify OSS controller that allow users to search download location regardless of https://, http://, www

## Type of change
Please insert 'x' one of the type of change.
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by soimkim** [comment]
Dear @doggai10 ,
Would you elaborate on what you were trying to fix?
And please share your test results.

**Comment by doggai10** [comment]
Dear @soimkim ,
what i was trying to fix was that check HttpServletRequest parameter homepage and convert url.  
if homepage url starts with http:// or www,  OSS search  couldn't find OSS that has exact same download location.  
so if homepage starts with http:// or www., convert url that starts with https:// and save homepage url using setHomepage function. 

- test results
스크린샷 2021-08-29 오후 2 22 46
스크린샷 2021-08-29 오후 2 22 30
스크린샷 2021-08-29 오후 2 22 12

File: src/main/java/oss/fosslight/controller/OssController.java [hunk 1]
```diff
@@ -145,7 +145,24 @@ public String list(HttpServletRequest req, HttpServletResponse res, Model model)
...
+			if(homepage.equals("http://")|| homepage.equals("www") || homepage.equals("")){
+				ossMaster.setHomepage("https://");
+			}else{
+				if(homepage.startsWith("http://")){
+					urls= homepage.split("://");
+				}
+				if(homepage.startsWith("www.")){
+					urls= homepage.split("www.");
+				}
+				ossMaster.setHomepage("https://"+urls[1]);
```
Comment (epicarts) [inline src/main/java/oss/fosslight/controller/OssController.java:162]: If `http://` or `www` is not in included, homepage variable is `https://null`. 

스크린샷 2021-08-31 오후 6 09 13

**Review by epicarts — COMMENTED** [review]
Dear @doggai10 , 
I tested this PR. and i found some issue.

스크린샷 2021-08-31 오후 6 35 31

It's not find anywhere. when http:// or www is not in included. 

스크린샷 2021-08-31 오후 6 37 31
but it can be searched on "demo.foosslight.org" site.

File: src/main/java/oss/fosslight/controller/OssController.java (hunk 1 above)
Comment (doggai10) [reply src/main/java/oss/fosslight/controller/OssController.java]: Dear @epicarts ,
first of all thank you for let me know about this.
i fixed the case that homepage url doesn't start with http or www .

- test results
스크린샷 2021-08-31 오후 10 40 45
스크린샷 2021-08-31 오후 10 28 35

**Comment by soimkim** [comment]
Dear @doggai10 ,
Please merge commits into one.

**Pull request closed without merging**